### PR TITLE
Fixes #3012

### LINKED ISSUE #3012: Paket restore silently fails when TargetFramework(s) are specified in Directory.Build.props and not csproj
### Description

When TargetFramework(s) are defined in Directory.Build.props Paket restore silently fails, it should be able to get it from msbuild at runtime.

### Repro steps

Have Directory.Build.props in solution root folder e.g.:

```xml
<Project>
 <Import Project=".\Versions.props"/>
  <PropertyGroup>
      <FullFrameworkVersion>net462</FullFrameworkVersion>
   </PropertyGroup>
</Project>
```

and then use the property to define TargetFramework in underlying project files, e.g.:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
   <TargetFramework>$(FullFrameworkVersion)</TargetFramework>
    <AssemblyName>SomeAssembly</AssemblyName>
...
  </PropertyGroup>
  <Import Project="..\..\.paket\Paket.Restore.targets" />
</Project>
```

### Expected behavior
**Paket restore** shouldn't silently fail. Ideally it should be able to get Framework information from msbuild from runtime.

### Actual behavior
**Paket restore** is not able to get TargetFramework(s) information from project file and silently fails.

### Known workarounds
Don't define TargetFramework in Directory.Build.props .
#3013

### PR DESCRIPTION
**Pull request by msatina**
fix #3012

Get TargetFramework(s) from runtime as a fallback in case they are not defined in csproj file.

### CODE REVIEW

**Comment by enricosada** [comment]
as a note, VS need to fix this scenario probably https://github.com/fsprojects/Paket/issues/2863#issuecomment-339365577

File: src/Paket.Core/embedded/Paket.Restore.targets [hunk 1]
```diff
@@ -69,9 +69,11 @@
       <PaketRestoreRequired Condition=" '$(PaketRestoreLockFileHash)' == '' ">true</PaketRestoreRequired>
     </PropertyGroup>
 
+
     <!-- Do a global restore if required -->
     <Exec Command='$(PaketBootStrapperCommand)' Condition="Exists('$(PaketBootStrapperExePath)') AND !(Exists('$(PaketExePath)'))" ContinueOnError="false" />
-    <Exec Command='$(PaketCommand) restore' Condition=" '$(PaketRestoreRequired)' == 'true' " ContinueOnError="false" />
```
Comment (matthid) [inline src/Paket.Core/embedded/Paket.Restore.targets]: Note that calling `paket restore` is by design here. We tweaked `paket restore` for maximum performance (including skipped command line parsing). With this change restore will most likely be a lot slower if stuff was already restored.

Comment (matthid) [reply src/Paket.Core/embedded/Paket.Restore.targets]: What I'm trying to say is that we either need a different solution / carefully tweak performance for this case or some other trick

**Comment by forki** [comment]
ping

**Comment by forki** [comment]
@MattDrivenDev better?

**Comment by forki** [comment]
thanks!

**Pull request merged**